src/infrasys/arrow_storage.py

```diff
@@ -127,14 +131,18 @@ def _get_single_time_series(
             normalization=metadata.normalization,
         )
 
+    def get_raw_time_series(self, time_series_uuid: UUID) -> NDArray:
+        fpath = self._ts_directory.joinpath(f"{time_series_uuid}{EXTENSION}")
+        with pa.memory_map(str(fpath), "r") as source:
+            base_ts = pa.ipc.open_file(source).get_record_batch(0)
+            logger.trace("Reading time series from {}", fpath)
+        return base_ts[str(time_series_uuid)].to_numpy()
```

Comment:
It looks like the old code only reads `length` entries of data from a memory-mapped object. This new code fully materializes the array into memory. Is that correct?

Reply:
The main reason I cast this to numpy is to be consistent in the return type of get_raw_time_series. This does however have the implication that the normal get_time_series is materializing numpy arrays because I refactored that function to use get_raw_timeseries. I could return the original function to return arrow arrays (or tables?). For the get_raw_data, the main use case is for the storage conversion, so I don't think it's an issue to materialize the data into memory as that is required to write to the new storage.

Reply:
Returning a numpy array seems good. Could you pass index and length to this function as optional parameters? If None, read the whole the file. Then the original behavior is preserved.